worker/sshserver/worker.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"context"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/state"
+)
+
+// sshServerWorker is a worker that runs an embedded SSH server.
+// See the server implementation for more details on how it works.
+type sshServerWorker struct {
+	tomb tomb.Tomb
+
+	srv *ssh.Server
+}
+
+// NewSSHServerWorker returns a new worker that runs an embedded SSH server.
+func NewSSHServerWorker(statePool *state.StatePool, jumpHostKey string) (worker.Worker, error) {
+	srv, err := NewSSHServer(statePool, jumpHostKey)
```

Comment:
This should be passed in as a config item. This will make testing of the worker simpler.

Reply:
What do you mean? (Sorry for asking questions a lot I'm still understanding Juju aha)